src/forms/inputs/file-input/helpers.js

```diff
@@ -0,0 +1,49 @@
+import { castArray } from '../../../utils'
+
+export async function readFiles ({ newFiles, existingFiles }) {
+  // Do not reload files that have been successfully loaded
+  const filesToLoad = newFiles.filter((file) => {
```

Comment:
I'd lift this comparison out of this function back into the component (could use a helper for it though)

```
const files = [ ...e.target.files ] // spread might not be necessary
const newFiles = removeExistingFiles(files, values)
const filesWithUrls = await readFiles(newFiles)
```

Reply:
I believe spread is necessary because `.files` is array-like (similar to arguments)

Reply:
😵gotcha.